core/src/main/java/com/kodebeagle/javaparser/MethodInvocationResolver.java

```diff
@@ -544,4 +590,84 @@ public String toString() {
                     '}';
         }
     }
+
+    public static class JavaDoc {
+
+        private String name;
+        private String comment;
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public String getComment() {
+            return comment;
+        }
+
+        public void setComment(String comment) {
+            this.comment = comment;
+        }
+
+        public JavaDoc(String name, String comment) {
+            this.name = name;
+            this.comment = comment;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            JavaDoc javaDoc = (JavaDoc) o;
+
+            return name.equals(javaDoc.name);
+
+        }
+
+        @Override
+        public int hashCode() {
+            return name.hashCode();
+        }
+
+        @Override
+        public String toString() {
+            return "JavaDoc{" +
+                    "name='" + name + '\'' +
+                    ", comment='" + comment + '\'' +
+                    '}';
+        }
+    }
+
+    public static class MethodJavadoc extends JavaDoc {
+
+        public MethodJavadoc(String name, String comment) {
+            super(name, comment);
+        }
+    }
+
+    public static class TypeJavadoc extends JavaDoc {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Override this superclass' "equals" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2160)